### PR TITLE
Align style doc with samples

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - enabled clang-format for code snippets in the documentation
 - Aligned the code style with samples

### Tickets:
 - NA

### CODE REVIEW

File: docs/snippets/movidius-programming-guide.cpp [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+    //! [part0]
 
-//! [part1]
-// numRequests is the number of frames (max size, equal to the number of VPUs in use)
-std::vector<Request> request(numRequests);
-//! [part1]
+    //! [part1]
+    // numRequests is the number of frames (max size, equal to the number of VPUs in use)
+    std::vector<Request> request(numRequests);
+    //! [part1]
```
Comment (ilya-lavrenov) [inline docs/snippets/movidius-programming-guide.cpp:19]: in this case all snippets in docs are shifted by 4 spaces..

Comment (ilyachur) [reply docs/snippets/movidius-programming-guide.cpp]: Yes good point.Fixed indentation for snippets

File: docs/snippets/InferenceEngine_network_with_state_infer.cpp [hunk 2]
```diff
@@ -2,108 +2,102 @@
...
-        }
-        
-        // 7. Initialize memory state before starting
-        for (auto &&state : inferRequest.QueryState()) {
-            state.Reset();
-        }
-
-        //! [part1]
-        // input data
-        std::vector<float> data = { 1,2,3,4,5,6};
```
Comment (ilyachur) [inline docs/snippets/InferenceEngine_network_with_state_infer.cpp:54]: @ilya-lavrenov Here we have an example with indentation

Comment (ilyachur) [reply docs/snippets/InferenceEngine_network_with_state_infer.cpp]: I have disabled clang-format for snippets

File: docs/template_extension/CMakeLists.txt [hunk 3]
```diff
@@ -11,6 +11,7 @@ find_package(ngraph REQUIRED OPTIONAL_COMPONENTS onnx_importer)
 find_package(InferenceEngine REQUIRED)
 find_package(OpenCV QUIET COMPONENTS core)
 
+file(GLOB_RECURSE template_extension_src "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/*.hpp")
```
Comment (ilya-lavrenov) [inline docs/template_extension/CMakeLists.txt:14]: let's add headers to `add_library`

File: docs/template_extension/CMakeLists.txt [hunk 4]
```diff
@@ -32,4 +33,5 @@ if (ngraph_onnx_importer_FOUND)
     target_link_libraries(${TARGET_NAME} PRIVATE ${ONNX_IMPORTER_LIBRARIES})
     target_compile_definitions(${TARGET_NAME} PRIVATE NGRAPH_ONNX_IMPORT_ENABLED)
 endif()
+add_clang_format_target(${TARGET_NAME}_clang FOR_SOURCES ${template_extension_src})
```
Comment (ilya-lavrenov) [inline docs/template_extension/CMakeLists.txt:36]: let's move beyond the `# [cmake:extension]`

File: docs/template_extension/CMakeLists.txt (hunk 3 above)
Comment (ilyachur) [reply docs/template_extension/CMakeLists.txt]: Fixed

File: docs/template_extension/CMakeLists.txt (hunk 4 above)
Comment (ilyachur) [reply docs/template_extension/CMakeLists.txt]: Fixed

**Comment by ilyachur** [comment]
@ilya-lavrenov Please take a look

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**